pkg/config/sync.go

```diff
@@ -245,12 +246,15 @@ func mergeConfig(from, to *controller.OperatorConfiguration) {
 		}
 		if from.Workspace.DefaultStorageSize != nil {
 			if to.Workspace.DefaultStorageSize == nil {
-				to.Workspace.DefaultStorageSize = make(map[string]resource.Quantity)
+				to.Workspace.DefaultStorageSize = &v1alpha1.StorageSizes{}
+
 			}
-			for storageClassName, PVCSize := range from.Workspace.DefaultStorageSize {
-				to.Workspace.DefaultStorageSize[storageClassName] = PVCSize
+			if from.Workspace.DefaultStorageSize.Common != nil {
+				to.Workspace.DefaultStorageSize.Common = from.Workspace.DefaultStorageSize.Common
+			}
+			if from.Workspace.DefaultStorageSize.PerWorkspace != nil {
+				to.Workspace.DefaultStorageSize.PerWorkspace = from.Workspace.DefaultStorageSize.PerWorkspace
```

Comment:
Since `resource.Quantity` isn't a base type, it's maybe safer to DeepCopy these fields rather than copying the reference
```suggestion
			if from.Workspace.DefaultStorageSize.Common != nil {
        commonSizeCopy := from.Workspace.DefaultStorageSize.Common.DeepCopy()
				to.Workspace.DefaultStorageSize.Common = &commonSizeCopy
			}
			if from.Workspace.DefaultStorageSize.PerWorkspace != nil {
        perWorkspaceSizeCopy := from.Workspace.DefaultStorageSize.PerWorkspace.DeepCopy()
				to.Workspace.DefaultStorageSize.PerWorkspace = &perWorkspaceSizeCopy
```